src/simulation/network/sim_network.rs

```diff
@@ -439,10 +452,21 @@ impl SimNetworkPartition {
         vehicle.advance_route_index();
         let link_id = vehicle.curr_link_id().unwrap();
         let link = links.get_mut(&link_id).unwrap();
-        events.publish_event(
```

Comment:
This might be more concise when using an `if let SimLink::Local(local) = link {...`

Just mentioning it because you said in your PR-Comment, that the downside of the changed logic is more cluttered code.